enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+Providing a new configuration API for the Insights should bring a better user experience as well as easier way to introduce new configuration options (an example could be a way how to run only particular gatherers, which should help when running in non-standard environments as for example HyperShift) and the operator status reporting. 
+
+## Motivation
+
+The current configuration of the Insights operator is available via the `support` secret, which has to be created first and the configuration options include at least 5 items (upload endpoint, gathering interval, config related to SCA pull down) at the moment and there will be more in the future. 
+Configuring all this stuff via secret is not a convenient way to configure a cluster operator. There is also a configuration option on how to exclude/include some of the gatherers, but this option is currently hardcoded in the image (i.e not exposed to the user).  Exposing these existing options and defining new ones will increase overall user experience. 
+This would also allow to expose the basic Insights analysis results in the corresponding status section. This functionality now relies on the in-cluster Prometheus (and it will probably not be fully available in the HyperShift architecture).
+Finally, one of the reasons for this enhancement is to define a new way for a user to trigger the Insights data collection.  
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Move the majority of the current configuration options to a new configuration API
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	Gathering Gathering `json:"gathering,omitempty"`
+	// SimpleContentAcces spec attribute includes all the configuration options related to
+	// periodical downloading of the SCA certificates.
+	SimpleContentAccess SimpleContentAccess `json:"simpleContentAccess,omitempty"`
+	// ClusterTransfer spec attribute includes all the configuration options related to
+	// periodical check of pending cluster transfers.
+	ClusterTransfer ClusterTransfer `json:"clusterTransfer,omitempty"`
+}
+
+type InsightsStatus struct {
```

Comment:
i would expect conditions used here

Reply:
Conditions? I am not sure what you mean.

Reply:
@mfojtik Are you basically suggesting to replace all the extra types (as `GatheringStatus`, `SimpleContentAccessStatus`, `ReportStatus`) by the [OperatorCondition](https://github.com/openshift/api/blob/354aa98a475c1fcd60b41aaee735da95d7318b42/operator/v1/types.go#L161) type (or any other similar type)? Or to define some "custom" condition type? Because I think I can't imagine how to convert e.g the `ReportStatus` (basically providing only numbers of healthchecks) to some type similar to a condition.

Reply:
I think I can remove (move to the CO conditions) `SimpleContentAccessStatus` and `ClusterTransferStatus` but I would keep the rest (`GatheringStatus` and `ReportStatus` - I think these two are not good candidates for the CO conditions)

Reply:
I removed the two mentioned above.